File: deps/v8/tools/testrunner/test_config.py
# Copyright 2018 the V8 project authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

from .utils import random_utils


class TestConfig(object):
  def __init__(self,
               command_prefix,
               extra_flags,
               framework_name,
               isolates,
               mode_flags,
               no_harness,
               noi18n,
               random_seed,
               run_skipped,
               shard_count,
               shard_id,
               shell_dir,
               target_os,
               timeout,
               verbose,
               regenerate_expected_files=False):
    self.command_prefix = command_prefix
    self.extra_flags = extra_flags
    self.framework_name = framework_name
    self.isolates = isolates
    self.mode_flags = mode_flags
    self.no_harness = no_harness
    self.noi18n = noi18n
    # random_seed is always not None.
    self.random_seed = random_seed or random_utils.random_seed()
    self.run_skipped = run_skipped
    self.shard_count = shard_count
    self.shard_id = shard_id
    self.shell_dir = shell_dir
    self.target_os = target_os
    self.timeout = timeout
    self.verbose = verbose
    self.regenerate_expected_files = regenerate_expected_files

File: deps/v8/test/js-perf-test/BigInt/test-config.js
// Copyright 2019 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

"use strict";

const TEST_ITERATIONS = 1000;
const SLOW_TEST_ITERATIONS = 50;
const BITS_CASES = [32, 64, 128, 256, 512, 1024, 2048, 4096, 8192];
const RANDOM_BIGINTS_MAX_BITS = 64 * 100;
